docs/_posts/2023-04-20-functional-modularity-2.md

```diff
@@ -3,56 +3,80 @@ title: "Modularity With Functional Programming Part 2"
 date: 2023-04-20 00:00:00 -0500
 excerpt: "Welcome to Modularity in Functional Programming! In this series, I will be addressing how using aspects of the functional paradigm can lead to program components that are easier to reuse and expand upon."
 classes: wide
-published: false
 ---
 
-In the first post of this series, we covered what modularity means and how functional programming is uniquely qualified to provide you modularity when writing programs. We covered examples from the "Gluing Functions Together" portion of John Hughes's 1984 memo titled "Why Functional Programming Matters." This second post will build on those ideas and cover two more examples from that section of the memo with accompanying Scala 3 sample code.
+In the first post of this series, we covered what modularity means and how functional programming is uniquely qualified to provide you modularity.
 
-While the first post demonstrated modularity with higher order functions, primarily with the higher order function foldRight and a list of single elements, these two examples extend that to show that the same manner of function composition can be done to perform operations on more complex data structures such as lists of lists and trees.
+We covered examples from the "Gluing Functions Together" portion of John Hughes's 1984 memo titled "Why Functional Programming Matters." This second post will build on those ideas and cover two more examples from that section of the memo with accompanying Scala 3 sample code.
 
-After this post, we will move on to the second idea Hughes presented in his memo, "Gluing Programs Together", which focuses on a feature common to many functional programming languages known as lazy evaluation.
+The two examples in this post extend the first post's "functions as glue" concept to show that the ideas of modularity through function composition and higher order functions can be used to perform operations on more complex data structures such as lists of lists and trees.
 
-If you missed the first post in this series, you can find it [here](/2023/03/functional-modularity-1/)), I recommend going back and reading that post as it explains the ideas these examples are expanding upon.
+Following this post the series will move on to the second idea Hughes presented in his memo, "Gluing Programs Together", which focuses on a feature common to many functional programming languages known as "lazy evaluation."
+
+If you missed the first post in this series, you can find it [here](/2023/03/functional-modularity-1/), I recommend going back and reading it if the ideas of modularity and higher order functions presented here are feeling unfamiliar.
```

Comment:
```suggestion
If you missed the first post in this series, you can find it [here]({% post_url 2023-03-29-functional-modularity-1 %}), I recommend going back and reading it if the ideas of modularity and higher order functions presented here are feeling unfamiliar.
```

Here I used the `post_url` tag to refer to your other post. This is helpful for ensuring things don't break locally, and will give you build-time support if you do something like change the name of the post.